docs/concepts/chain-key-design.md

```diff
@@ -0,0 +1,211 @@
+# Chain Key Design
+
+## Introduction to GalaChain Chain Keys
+
+In GalaChain, the `@ChainKey` decorator is a powerful feature for defining structured composite keys that organize data in the underlying LevelDB key-value store. These composite keys enable efficient querying and data retrieval patterns within your blockchain application.
+
+## What Are Chain Keys?
+
+Chain Keys are ordered collections of property values that together form a unique composite key for storing state data. They leverage Hyperledger Fabric's native composite key capabilities but provide a structured, type-safe approach through GalaChain's decorators.
+
+## Basic Usage
+
+```typescript
+import { ChainObject, ChainKey } from '@galachain/sdk';
+
+export class PlayerInventoryItem extends ChainObject {
```

Comment:
there is also the mandatory static `INDEX_KEY` property which defines a class/category (?) of objects. It should be unique per class across the chaincode